ql/src/java/org/apache/hadoop/hive/ql/Context.java

```diff
@@ -256,7 +256,8 @@ public String toString() {
   public enum RewritePolicy {
 
     DEFAULT,
-    ALL_PARTITIONS;
+    ALL_PARTITIONS,
+    PARTITION;
```

Comment:
why do you need separate policy for PARTITION? not ALL_PARTITIONS isn't enough?

Reply:
We can manage without 'PARTITION' rewrite policy, but to me it is confusing why we have the value 'ALL_PARTITIONS' for compacting all partitions, but there is no policy when compacting one partition only and it seems to be unclear to others too based on this comment https://github.com/apache/hive/pull/5248#discussion_r1621812294. 

Addition of 'PARTITION' rewrite policy also simplifies code in a few places. For example:
This was in `validateSinkDesc` previously:
```
      if (RewritePolicy.fromString(conf.get(ConfVars.REWRITE_POLICY.varname, RewritePolicy.DEFAULT.name())) ==
          RewritePolicy.ALL_PARTITIONS || conf.get(CompactorContext.COMPACTION_PART_SPEC_ID) != null) {
        // Table rewriting has special logic as part of IOW that handles the case when table had a partition evolution
        return;
      }
```
And now:
```
      if (RewritePolicy.fromString(conf.get(ConfVars.REWRITE_POLICY.varname, RewritePolicy.DEFAULT.name())) !=
          RewritePolicy.DEFAULT) {
        // Table rewriting has special logic as part of IOW that handles the case when table had a partition evolution
        return;
      }
```

Reply:
ALL_PARTITIONS was added to distinguish from IOW

Reply:
Removed RewritePolicy.PARTITION and instead added CompactionLevel enum which is needed in a few places to pass information that partition level request is being handled.